wgengine/magicsock/magicsock.go

```diff
@@ -943,6 +963,10 @@ func (c *Conn) determineEndpoints(ctx context.Context) ([]tailcfg.Endpoint, erro
 	// re-run.
 	eps = c.endpointTracker.update(time.Now(), eps)
 
+	for _, ap := range c.staticEndpoints.AsSlice() {
+		addAddr(ap, tailcfg.EndpointExplicitConf)
```

Comment:
I have also placed this below the bit that [caches endpoints](https://github.com/tailscale/tailscale/blob/v1.70.0/wgengine/magicsock/magicsock.go#L944) as I understand we don't need to cache these and if we do, they don't get removed from advertised endpoints on update. We do cache endpoints provided via the env var, but I think it doesn't really matter as those cannot be updated.